src/com/ms/silverking/cloud/config/HostGroupTable.java

```diff
@@ -23,234 +23,237 @@
-            if (hostGroups.contains(serverHostGroup)) {
-                return true;
-            }
-        }
-        return false;
+    return hosts.build();
+  }
+
+  public Set<InetAddress> getHosts(String groupName) {
+    Set<InetAddress> s;
+
+    s = groupToServerMap.getSet(groupName);
+    if (s == null) {
+      s = ImmutableSet.of();
     }
-    
-    public Set<String> getHostGroups(String server) {
-        try {
-            Set<String> hostGroups;
-            
-            hostGroups = getHostGroups(InetAddress.getByName(server));
-            if (hostGroups == null) {
-                hostGroups = ImmutableSet.of();
-            }
-            return hostGroups;
-        } catch (UnknownHostException e) {
-            throw new RuntimeException(e);
-        }
+    return s;
+  }
+
+  public Set<String> getHostAddresses(String groupName) {
+    Set<String> s;
+
+    s = groupToServerAddressMap.getSet(groupName);
+    if (s == null) {
+      s = ImmutableSet.of();
     }
-    
-    public Set<String> getHostGroups(InetAddress server) {
-        return serverToGroupMap.getSet(server);
+    return s;
+  }
+
+  public boolean serverInHostGroupSet(String server, Set<String> hostGroups) {
+    Set<String> serverHostGroups;
+
+    serverHostGroups = getHostGroups(server);
+    for (String serverHostGroup : serverHostGroups) {
+      if (hostGroups.contains(serverHostGroup)) {
+        return true;
+      }
     }
-    
-    @Override
-    public long getVersion() {
-        return version;
+    return false;
+  }
+
+  public Set<String> getHostGroups(String server) {
+    try {
+      Set<String> hostGroups;
+
+      hostGroups = getHostGroups(InetAddress.getByName(server));
+      if (hostGroups == null) {
+        hostGroups = ImmutableSet.of();
+      }
+      return hostGroups;
+    } catch (UnknownHostException e) {
+      throw new RuntimeException(e);
     }
-    
-    public static HostGroupTable parse(String def, long version) {
-        try {
-            return parse(new ByteArrayInputStream(def.getBytes()), version);
-        } catch (IOException ioe) {
-            throw new RuntimeException("Unexpected exception", ioe);
-        }
+  }
+
+  public Set<String> getHostGroups(InetAddress server) {
+    return serverToGroupMap.getSet(server);
+  }
+
+  @Override
+  public long getVersion() {
+    return version;
+  }
+
+  public static HostGroupTable parse(String def, long version) {
+    try {
+      return parse(new ByteArrayInputStream(def.getBytes()), version);
+    } catch (IOException ioe) {
+      throw new RuntimeException("Unexpected exception", ioe);
     }
-    
-    public static HostGroupTable parse(File fileName, long version) throws IOException {
-        return parse(new FileInputStream(fileName), version);
+  }
+
+  public static HostGroupTable parse(File fileName, long version) throws IOException {
+    return parse(new FileInputStream(fileName), version);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the resource created by the following constructor call: *FileInputStream*. Currently, there are execution paths that do not contain closure statements, for example, when HostGroupTable.parse throws an exception. Either a) close the object returned by *FileInputStream()* in a try-finally block or b) close the resource by declaring the object returned by *FileInputStream()* in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).